nominal/core/asset.py

```diff
@@ -229,6 +230,36 @@ def list_data_scopes(self) -> Sequence[tuple[str, ScopeType]]:
         """
         return (*self.list_datasets(), *self.list_connections(), *self._list_logsets(), *self.list_videos())
 
+    def get_or_create_dataset(
+        self,
+        *,
+        data_scope_name: str,
+        name: str | None = None,
+        description: str | None = None,
+        labels: Sequence[str] = (),
+        properties: Mapping[str, str] | None = None,
+    ) -> Dataset:
+        """Retrieve a dataset by data scope name, or create a new one if it does not exist."""
+        try:
+            return self.get_dataset(data_scope_name)
+        except ValueError:
+            enriched_dataset = self._clients.catalog.create_dataset(
+                self._clients.auth_header,
+                scout_catalog.CreateDataset(
+                    name=name or data_scope_name,
+                    description=description,
+                    properties={} if properties is None else dict(properties),
+                    labels=list(labels),
+                    is_v2_dataset=True,
+                    metadata={},
+                    origin_metadata=scout_catalog.DatasetOriginMetadata(),
+                    workspace=self._clients.workspace_rid,
+                ),
+            )
+            dataset = Dataset._from_conjure(self._clients, enriched_dataset)
```

Comment:
hmmm I'd like to just reuse `client.create_dataset` rather than duplicate the impl, but we don't have that method available here. kinda surprised this is the first time we've had that issue...

let's make a private utility func at least that they can both depend on